src/testdir/test_functions.vim

```diff
@@ -950,29 +950,40 @@ func Test_reg_executing_and_recording()
 endfunc
 
 func Test_libcall_libcallnr()
-  " TODO:
-  " - it should be possible to test libcall()/libcallnr() on Windows.
-  " - on macOs, the test will not find /lib64/libc.so.[1-9] so the test
-  "   will be skipped. Can we test libcall()/libcallnr() on macOs?
-  if !has('unix') || !has('libcall')
+  if !has('libcall')
     return
   endif
 
-  let libs = split(glob('/lib/libc.so.[1-9]'), "\n")
-        \  + split(glob('/lib64/libc.so.[1-9]'), "\n")
-  if len(libs) > 0
-    let libc = libs[-1]
+  if has('win32')
+    let libc = 'msvcrt.dll'
+  else
+    " TODO: on macOs, the test will not find /lib64/libc.so.[1-9] so the test
+    " will be skipped. Can we test libcall()/libcallnr() on macOs?
+    let libs = split(glob('/lib/libc.so.[1-9]'), "\n")
+          \  + split(glob('/lib64/libc.so.[1-9]'), "\n")
+    if len(libs) > 0
+      let libc = libs[-1]
+    endif
+  endif
 
-    let home = libcall(libc, 'getenv', 'HOME')
-    call assert_equal($HOME, home)
+  if len(libc) == 0
+    return
+  endif
 
-    let pid = libcallnr(libc, 'getpid', '')
-    call assert_equal(getpid(), pid)
+  let $X_TEST_ENV_VAR_ = 'foo'
+  call assert_equal('foo', libcall(libc, 'getenv', 'X_TEST_ENV_VAR_'))
```

Comment:
This might not work on Windows.
If Vim is compiled with a recent version of MSVC, it will be linked with a newer version of msvcrXXX.dll (or other CRT dll) instead of msvcrt.dll.
Therefore `let $FOO='bar'` doesn't affect the return value of `getenv()` from msvcrt.dll.

Reply:
Thanks. I'll do it how you suggested earlier than, i.e. by using the USERPROFILE
environment variable. The reason I tried to do it differently, is that I was hoping that
my solution would be identical on all operating systems.  Clearly it's not.

I don't have my development machine right now so I'll fix the Windows test failure later today.